tsdb/chunkenc/histo.go

```diff
@@ -258,9 +258,84 @@ func (a *HistoAppender) Appendable(h histogram.SparseHistogram) ([]Interjection,
 	if !ok {
 		return nil, nil, false
 	}
+	if counterReset(a.posbuckets, h.PositiveBuckets, a.posSpans, h.PositiveSpans) {
```

Comment:
We also need to check h.ZeroCount and h.Count. In fact, almost all counter resets should be caught be either the `compareSpans` above (because buckets disappeared) or be `h.Count` going down. (The rare case where that doesn't work is if the histogram has received more observations after the reset than it had when it was last scraped _and_ exactly the same bucket layout has been reached.)

So I think we should check in this order:
1. h.Count (catches most)
2. h.ZeroCount (rare, but cheap to check)
3. Only then do the expensive check of buckets.